sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/OperatorsShouldBeOverloadedConsistently.cs

```diff
@@ -0,0 +1,146 @@
+
+            return effectiveAccessibility == Accessibility.Public ||
+                effectiveAccessibility == Accessibility.Protected ||
+                effectiveAccessibility == Accessibility.ProtectedOrInternal;
+        }
+
+        private static IEnumerable<string> FindMissingMethods(INamedTypeSymbol classSymbol)
+        {
+            var implementedMethods = GetImplementedMethods(classSymbol).ToHashSet();
+            var requiredMethods = new HashSet<string>();
+
+            if (implementedMethods.Contains(MethodName.OperatorPlus))
+            {
+                requiredMethods.Add(MethodName.OperatorMinus);
+                requiredMethods.Add(MethodName.OperatorEquals);
+                requiredMethods.Add(MethodName.OperatorNotEquals);
+                requiredMethods.Add(MethodName.ObjectEquals);
+                requiredMethods.Add(MethodName.ObjectGetHashCode);
+            }
+
+            if (implementedMethods.Contains(MethodName.OperatorMinus))
+            {
+                requiredMethods.Add(MethodName.OperatorPlus);
+                requiredMethods.Add(MethodName.OperatorEquals);
+                requiredMethods.Add(MethodName.OperatorNotEquals);
+                requiredMethods.Add(MethodName.ObjectEquals);
+                requiredMethods.Add(MethodName.ObjectGetHashCode);
+            }
+
+            if (implementedMethods.Contains(MethodName.OperatorEquals))
+            {
+                requiredMethods.Add(MethodName.OperatorNotEquals);
+                requiredMethods.Add(MethodName.ObjectEquals);
+                requiredMethods.Add(MethodName.ObjectGetHashCode);
+            }
+
+            if (implementedMethods.Contains(MethodName.OperatorNotEquals))
+            {
+                requiredMethods.Add(MethodName.OperatorEquals);
+                requiredMethods.Add(MethodName.ObjectEquals);
+                requiredMethods.Add(MethodName.ObjectGetHashCode);
+            }
+
+            return requiredMethods.Except(implementedMethods);
+        }
+
+        private static IEnumerable<string> GetImplementedMethods(INamedTypeSymbol classSymbol)
+        {
+            var classMethods = classSymbol
+                .GetMembers()
+                .OfType<IMethodSymbol>()
+                .Where(m => !m.IsConstructor())
+                .ToList();
+
+            if (classMethods.Any(KnownMethods.IsOperatorBinaryPlus)) yield return MethodName.OperatorPlus;
+            if (classMethods.Any(KnownMethods.IsOperatorBinaryMinus)) yield return MethodName.OperatorMinus;
+            if (classMethods.Any(KnownMethods.IsOperatorEquals)) yield return MethodName.OperatorEquals;
+            if (classMethods.Any(KnownMethods.IsOperatorNotEquals)) yield return MethodName.OperatorNotEquals;
+
+            if (classMethods.Any(KnownMethods.IsObjectEquals)) yield return MethodName.ObjectEquals;
+            if (classMethods.Any(KnownMethods.IsObjectGetHashCode)) yield return MethodName.ObjectGetHashCode;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add curly braces around the nested statement(s) in this 'if' block. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS121)